Realm/RLMObservation.mm

```diff
@@ -521,8 +529,12 @@ void RLMWillChange(std::vector<realm::BindingContext::ObserverState> const& obse
         NSMutableIndexSet *indexes = [NSMutableIndexSet new];
         for (auto const& o : observed) {
             forEach(o, [&](realm::ColKey colKey, auto const& change, RLMObservationInfo *info) {
-                info->willChange(info->columnName(colKey),
-                                 convert(change.kind), convert(change.indices, indexes));
+                if (colKey.is_set()) {
```

Comment:
Is this conditional actually needed? It's just ensuring that the second parameter is `nil` even if `change.indices` isn't empty for a set, but if we aren't going to be using `change.indices` for sets then we shouldn't be populating it in the first place.

Reply:
The reason why this conditional is here is because the `forEach` iterates over the changes in `ObserverState` and passes back the `ColKey` allowing us to differentiate between collection types. I couldn't see any other way to avoid passing the populated indices without this.

Reply:
The code which is populating ObserverState should be skipping sets rather than going through the work of calculating the change indices and then just discarding them.

Reply:
I'll need to change this from the core side. Should this PR be blocked until that is done, or could it be placed in with Dictionary when that PR is available?

Reply:
https://github.com/realm/realm-core/pull/4422/files

Reply:
I merged in the above, but it will need a core release so the xcproj can consume the changes